blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,6 +147,13 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            long current = System.currentTimeMillis();
+
+            // valid if collectionTime is greater than past window or less than future window
+            return ((collectionTime >= current - pastDiff ) && ( collectionTime <= current + futureDiff ));
```

Comment:
combined the isValid() and getValidationErrors() into one new method validate();  Which will only be call once and do a validation on the time and add errors to a returnable errors list.  Convert JSONMetricsContainer to process json when created/initializing, which validates and populate the errors list of any validation failure.